Documentation/network/concepts/ipam/multi-pool.rst

```diff
@@ -0,0 +1,153 @@
+     allocate the pod's IP from the pool named ``A``.
+  2. If there is no annotation on the pod, but the namespace of the pod has an
+     ``ipam.cilium.io/ip-pool=B`` annotation, Cilium will
+     allocate the pod's IP from the pool named ``B``.
+  3. If neither the pod nor the namespace have a ``ipam.cilium.io/ip-pool`` annotation,
+     the pod's IP will be allocated from the pool named ``default``.
+
+The annotation is only considered when the pod is created. Changing the ``ip-pool``
+annotation on an already running pod has no effect.
+
+The ``CiliumNode`` resource is extended with an additional ``spec.ipam.pools`` section:
+
+``spec.ipam.pools.requested``
+  List of IPAM pool requests for this node. Each entry specifies the pool and the number of
+  requested IP addresses. This field is owned and written to by the Cilium agent running on
+  the particular node. It is read by the Cilium operator to fulfill the requests.
+
+``spec.ipam.pools.allocated``
+  List of CIDRs allocated to this node and the pool they were allocated from.
+  Cilium operator adds new PodCIDRs to this field. Cilium agent removes PodCIDRs
+  it has released and is no longer using.
+
+IP pools themselves are managed using the cluster-wide ``CiliumPodIPPool`` custom resource.
+Each ``CiliumPodIPPool`` contains the cluster-wide CIDR from which per-node PodCIDRs
+are allocated from:
+
+.. code-block:: yaml
+
+  apiVersion: cilium.io/v2alpha1
+  kind: CiliumPodIPPool
+  metadata:
+    name: green-pool
+  spec:
+    ipv4:
+      cidrs:
+        - 10.20.0.0/16
+        - 10.30.0.0/16
+      maskSize: 24
+    ipv6:
+      cidrs:
+        - fd00::/104
+      maskSize: 120
+
+New pools can be added at run-time. The list of CIDRs in each pool can also be
+extended at run-time. In-use CIDRs may not be removed from an existing pool, and
+existing pools may not be deleted if they are still in use by a Cilium node.
+The mask size of a pool is immutable and the same for all nodes.
+
+Configuration
+*************
+
+Multi-Pool IPAM can be enabled using the ``ipam.mode=multi-pool`` Helm value.
+To have the Cilium operator automatically create ``CiliumPodIPPools`` custom
+resources at startup, use the ``ipam.operator.autoCreateCiliumPodIPPools`` Helm
+value. It contains a map which follows the ``CiliumPodIPPools`` CRD schema
+described above.
+
+.. code-block:: yaml
+
+  ipam:
+    mode: multi-pool
+    operator:
+      autoCreateCiliumPodIPPools:
+        default:
+          ipv4:
+            cidrs:
+              - 10.10.0.0/16
+            maskSize: 24
+        other:
+          ipv4:
+            cidrs:
+              - 10.20.0.0/16
+            maskSize: 24
+
+.. note::
+
+  For a practical tutorial on how to enable this mode in Cilium, see
+  :ref:`gsg_ipam_crd_multi_pool`.
+
+Allocation Parameters
+---------------------
+
+Cilium agent can be configured to pre-allocate a minimum number of IPs from each
```

Comment:
I've dropped the term "minimum" from this section because I think it is not entirely correct anyway. For the purpose of explaining the pre-allocation algorithm I don't think it adds value. The exact mechanism is already documented in the formula below.

Reply:
Looks great, thank you! There is still one usage of `at least` in this context on line 117. Can we remove that one as well?

Reply:
Dropped this as well, thanks.